python/feldera/rest/client.py

```diff
@@ -0,0 +1,455 @@
+
+        body = {
+            "config": pipeline.config,
+            "description": pipeline.description or "",
+            "connectors": pipeline.attached_connectors,
+            "program_name": pipeline.program_name,
+        }
+
+        resp = self.http.put(
+            path=f"/pipelines/{pipeline.name}",
+            body=body,
+        )
+
+        pipeline.id = resp.get("pipeline_id")
+
+    def get_pipeline_config(self, name: str) -> dict:
+        """
+        Get the configuration of a pipeline by name
+        """
+
+        resp = self.http.get(
+            path=f"/pipelines/{name}/config",
+        )
+
+        return resp
+
+    def validate_pipeline(self, name: str) -> bool:
+        """
+        Validate a pipeline.
+        Checks whether the pipeline is configured correctly.
+        This includes checking whether the pipeline references a valid compiled program,
+        whether the connectors reference valid tables/views in the program, and more.
+        """
+
+        resp = self.http.get(
+            path=f"/pipelines/{name}/validate",
+        )
+
+        return "success" in resp
+
+    def delete_pipeline(self, name: str):
+        """
+        Deletes a pipeline by name
+        :param name: The name of the pipeline
+        """
+        resp = self.http.delete(
+            path=f"/pipelines/{name}",
+        )
+
+    def get_pipeline_stats(self, name: str) -> dict:
+        """
+        Get the pipeline metrics and performance counters
+        """
+
+        resp = self.http.get(
+            path=f"/pipelines/{name}/stats",
+        )
+
+        return resp
+
+    def start_pipeline(self, pipeline_name: str):
+        """
+        Start a pipeline
+        :param pipeline_name: The name of the pipeline to start
+        """
+        self.http.post(
+            path=f"/pipelines/{pipeline_name}/start",
+        )
+
+        while True:
+            status = self.get_pipeline(pipeline_name).state.get("current_status")
+
+            if status == "Running":
+                break
+            elif status == "Failed":
+                raise RuntimeError(f"Failed to start pipeline")
+
+            logging.debug("still starting %s, waiting for 100 more milliseconds", pipeline_name)
+            time.sleep(0.1)
+
+    def pause_pipeline(self, pipeline_name: str):
+        """
+        Stop a pipeline
+        :param pipeline_name: The name of the pipeline to stop
+        """
+        self.http.post(
+            path=f"/pipelines/{pipeline_name}/pause",
+        )
+
+        while True:
+            status = self.get_pipeline(pipeline_name).state.get("current_status")
+
+            if status == "Paused":
+                break
+            elif status == "Failed":
+                raise RuntimeError(f"Failed to pause pipeline")
+
+            logging.debug("still pausing %s, waiting for 100 more milliseconds", pipeline_name)
+            time.sleep(0.1)
+
+    def shutdown_pipeline(self, pipeline_name: str):

```

Comment:
timeout/check_interval/mentioning current duration in debug log as for starting a pipeline